src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketScmHelper.java

```diff
@@ -58,4 +59,28 @@ public BitbucketRepository getRepository(String projectName, String repositoryNa
             return new BitbucketRepository(-1, repositoryName, null, new BitbucketProject(projectName, null, projectName), repositoryName, RepositoryState.AVAILABLE);
         }
     }
+    
+    public BitbucketDefaultBranch getDefaultBranch(String projectName, String repositoryName) {
+        if (isBlank(projectName) || isBlank(repositoryName)) {
+            LOGGER.info("Error creating the Bitbucket SCM: The projectName and repositoryName must not be blank");
+            return new BitbucketDefaultBranch("", "", "", "", "", false);
```

Comment:
We return an empty repository in the `getRepository` method because we have to persist _something_ as part of the SCM initialization. That's not the case here, the branch is thrown away as soon as the branch is checked, so perhaps we could make this nullable and perform a presence/absence check more explicitly in the `BitbucketSCMSource.retrieveActions` method?

Reply:
Understood and implemented accordingly.